src/sqlparser/src/parser.rs

```diff
@@ -3581,15 +3581,17 @@ impl Parser {
     pub fn parse_set(&mut self) -> Result<Statement, ParserError> {
         let modifier = self.parse_one_of_keywords(&[Keyword::SESSION, Keyword::LOCAL]);
         if self.parse_keywords(&[Keyword::TIME, Keyword::ZONE]) {
-            let value = if self.parse_keyword(Keyword::DEFAULT) {
-                SetTimeZoneValue::Default
-            } else if self.parse_keyword(Keyword::LOCAL) {
-                SetTimeZoneValue::Local
-            } else if let Ok(ident) = self.parse_identifier() {
-                SetTimeZoneValue::Ident(ident)
-            } else {
-                let value = self.parse_value()?;
```

Comment:
`parse_value()` after a failed `parse_identifier()` does not work because the token position is not restored with `prev_token()`. But instead of using `prev_token()`, we follow the same pattern as `parse_set_variable`: use one `peek_xxx` and one `parse_xxx`